### PR TITLE
Feature/it 12111 enable default customer segment in srr

### PR DESCRIPTION
**Pull request by khadimRaath**
https://ekomi01.atlassian.net/browse/IT-12111
Plentymarkets-Plugin enable default project in srr while saving configurations.

**Changes:**
- Added functionality to enable default customer segment in SRR
- Code Optimization

### CODE REVIEW

File: src/Containers/EkomiFeedbackSmartWidget.php [hunk 1]
```diff
@@ -5,34 +5,42 @@
...
 /**
  * Ekomi Feedback Reviews Container
  */
-class EkomiFeedbackSmartWidget {
-
-    public function call(Twig $twig, $arg) {
-
+class EkomiFeedbackSmartWidget
+{
+
```
Comment (ghulammurtaza) [inline src/Containers/EkomiFeedbackSmartWidget.php]: extra space, please remove it

File: src/Helper/ConfigHelper.php [hunk 2]
```diff
@@ -7,79 +7,123 @@
...
+     */
     public function getPlentyIDs()
     {
         $plentyIDs = false;
-
-        $IDs = $this->config->get('EkomiFeedback.plenty_IDs');
-
-        $IDs = preg_replace('/\s+/', '', $IDs);
-
+        $IDs       = $this->config->get('EkomiFeedback.plenty_IDs');
```
Comment (ghulammurtaza) [inline src/Helper/ConfigHelper.php]: should be $Ids

File: src/Helper/ConfigHelper.php [hunk 3]
```diff
@@ -7,79 +7,123 @@
...
 
+    /**
+     * Gets Order Statuses array from plugin configurations.
+     *
+     * @return array
+     */
     public function getOrderStatus()
     {
-        $status = $this->config->get('EkomiFeedback.order_status');
+        $status      = $this->config->get('EkomiFeedback.order_status');
```
Comment (ghulammurtaza) [inline src/Helper/ConfigHelper.php]: please remove extra space before =

File: src/Helper/ConfigHelper.php [hunk 4]
```diff
@@ -88,28 +132,52 @@ public function getReferrerIds()
...
-	}
-
-	public function getPrcWidgetToken()
-	{
-		return $this->config->get('EkomiFeedback.prc_widget_token');
-	}
+    /**
+     * Gets Smart Check from plugin configurations.
+     *
+     * @return mixed
```
Comment (ghulammurtaza) [inline src/Helper/ConfigHelper.php]: please specify the return type(s) instead of mixed

File: src/Helper/EkomiHelper.php [hunk 5]
```diff
@@ -10,67 +10,97 @@
...
+     * @param ItemImageRepositoryContract        $imagesRepo
+     * @param CountryRepositoryContract          $countryRepo
+     */
+    public function __construct(WebstoreRepositoryContract $webStoreRepo, ConfigHelper $configHelper, ItemImageRepositoryContract $imagesRepo, CountryRepositoryContract $countryRepo)
+    {
         $this->configHelper = $configHelper;
         $this->webStoreRepo = $webStoreRepo;
-        $this->imagesRepo = $imagesRepo;
-        $this->countryRepo = $countryRepo;
+        $this->imagesRepo   = $imagesRepo;
```
Comment (ghulammurtaza) [inline src/Helper/EkomiHelper.php]: extra space in both lines

File: src/Helper/EkomiHelper.php [hunk 6]
```diff
@@ -10,67 +10,97 @@
...
+    public function preparePostVars($order)
+    {
+        $id       = $order['id'];
+        $plentyId = $order['plentyId'];
+        $fields   = array(
+            'shop_id'            => $this->configHelper->getShopId(),
+            'interface_password' => $this->configHelper->getShopSecret(),
+            'mode'               => $this->configHelper->getMode(),
+            'product_reviews'    => $this->configHelper->getProductReviews(),
+            'plugin_name'        => 'plentymarkets',
```
Comment (ghulammurtaza) [inline src/Helper/EkomiHelper.php]: please take plugin name as constant.

File: src/Helper/EkomiHelper.php [hunk 7]
```diff
@@ -10,67 +10,97 @@
...
+        );
+
+        $order['senderName']  = $this->getWebStoreName($plentyId);
         $order['senderEmail'] = '';
-
-	    foreach ($order['addresses'] as $key=>$address) {
-	        $countryInfo = $this->countryRepo->getCountryById($address['countryId']);
-
+        foreach ($order['addresses'] as $key=>$address) {
+            $countryInfo                             = $this->countryRepo->getCountryById($address['countryId']);
```
Comment (ghulammurtaza) [inline src/Helper/EkomiHelper.php]: extra space

File: src/Helper/ConfigHelper.php (hunk 2 above)
Comment (khadimRaath) [reply src/Helper/ConfigHelper.php]: fixed

File: src/Helper/ConfigHelper.php (hunk 3 above)
Comment (khadimRaath) [reply src/Helper/ConfigHelper.php]: fixed

File: src/Helper/ConfigHelper.php (hunk 4 above)
Comment (khadimRaath) [reply src/Helper/ConfigHelper.php]: fixed

File: src/Helper/EkomiHelper.php (hunk 5 above)
Comment (khadimRaath) [reply src/Helper/EkomiHelper.php]: fixed

File: src/Helper/EkomiHelper.php (hunk 6 above)
Comment (khadimRaath) [reply src/Helper/EkomiHelper.php]: Fixed

File: src/Helper/EkomiHelper.php (hunk 7 above)
Comment (khadimRaath) [reply src/Helper/EkomiHelper.php]: fixed

File: src/Helper/EkomiHelper.php [hunk 8]
```diff
@@ -29,63 +54,92 @@ public function __construct(WebstoreRepositoryContract $webStoreRepo, ConfigHelp
...
+     *
      * @return string
      */
-    public function getItemImageUrl($itemId) {
-        $images = $this->imagesRepo->findByItemId($itemId);
-        if (isset($images[0])) {
-            return $images[0]['url'];
+    protected function getWebStoreName($plentyId)
+    {
+        $temp1 = $this->webStoreRepo->findByPlentyId($plentyId)->toArray();
```
Comment (mrehanali) [inline src/Helper/EkomiHelper.php:100]: please use meaningful names for variables.

File: src/Helper/EkomiHelper.php [hunk 9]
```diff
@@ -164,4 +194,18 @@ protected function getStoreDomain($plentyId) {
...
+     * @param int $turnaroundTime
+     *
+     * @return array
+     */
+    public function prepareFilter($turnaroundTime)
+    {
+        $updatedAtFrom = date('Y-m-d\TH:i:s+00:00', strtotime("-{$turnaroundTime} day"));
+        $updatedAtTo = date('Y-m-d\TH:i:s+00:00');
+
+        return ['updatedAtFrom' => $updatedAtFrom, 'updatedAtTo' => $updatedAtTo];
```
Comment (mrehanali) [inline src/Helper/EkomiHelper.php:209]: please be consistent either use array() or [] on all places.

**Review by mrehanali — APPROVED** [review]

**Pull request merged**